clippy_lints/src/wildcard_imports.rs

```diff
@@ -102,21 +103,40 @@ declare_clippy_lint! {
 pub struct WildcardImports {
     warn_on_all: bool,
     test_modules_deep: u32,
+    ignore: bool,
 }
 
 impl WildcardImports {
     pub fn new(warn_on_all: bool) -> Self {
         Self {
             warn_on_all,
             test_modules_deep: 0,
+            ignore: false,
         }
     }
 }
 
 impl_lint_pass!(WildcardImports => [ENUM_GLOB_USE, WILDCARD_IMPORTS]);
 
 impl LateLintPass<'_> for WildcardImports {
+    fn check_mod(&mut self, cx: &LateContext<'_>, module: &Mod<'_>, _: HirId) {
```

Comment:
This needs a `check_mod_post`, that sets the `self.ignore` to `false` again. Otherwise, once checking one module that ends with `test.rs`, you'll never lint again in any other module. 

Aaaand I just noticed that there is no `check_mod_post`. Hmmm... 

This should also only be done on a file level. But a file is implicitly either a module itself AFAIK or `lib.rs`/`main.rs`. So checking the mod name should be enough.

Reply:
You probably want to do all that with `check_item` and `check_item_post`. You should be able to get the `Mod` name through the `Item` `ident`.

Reply:
Checking the mod name doesn't work for files because files are not items (sadly).
It seems like the only way to check that the file name isn't `test.rs` or `tests.rs` is to directly check the filename. As there are no "module_name" function.

Reply:
I can't seem to get a way to get the crate name from `check_item` (or even `check_crate`), I've been trying things for 5 or 6 hours now.
Would checking for `test` in any part of the path fix this issue?